clients/venice-admin-tool/src/main/java/com/linkedin/venice/datarecovery/DataRecoveryExecutor.java

```diff
@@ -0,0 +1,103 @@
+package com.linkedin.venice.datarecovery;
+
+import java.io.Console;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+/**
+ * DataRecoveryExecutor is the engine to run tasks in data recovery.
+ */
+public class DataRecoveryExecutor {
+  private final Logger LOGGER = LogManager.getLogger(DataRecoveryExecutor.class);
+  private final static int DEFAULT_POOL_SIZE = 10;
+  private final static int DEFAULT_POOL_TIMEOUT_IN_SECONDS = 30;
+  private final int poolSize;
+  private final ExecutorService pool;
+  private List<DataRecoveryTask> tasks;
+
+  public DataRecoveryExecutor() {
+    this(DEFAULT_POOL_SIZE);
+  }
+
+  public DataRecoveryExecutor(int poolSize) {
+    this.poolSize = poolSize;
+    this.pool = Executors.newFixedThreadPool(this.poolSize);
+  }
+
+  public void perform(Set<String> storeNames, StoreRepushCommand.Params params) {
+    String pass = getUserCredentials();
+    if (pass == null) {
+      LOGGER.error("Cannot get password, exit");
+      return;
+    }
+    params.setPassword(pass);
+
+    tasks = buildTasks(storeNames, params);
+    List<CompletableFuture<Void>> taskFutures = tasks.stream()
+        .map(dataRecoveryTask -> CompletableFuture.runAsync(dataRecoveryTask, pool))
+        .collect(Collectors.toList());
+    taskFutures.stream().map(CompletableFuture::join).collect(Collectors.toList());
+    displayTaskResult();
+
+    shutdownAndAwaitTermination();
```

Comment:
What is the intention of this function?
So we want to close the `executor` after submitting the recovery tasks, right?
If there is a big list, the recovery tasks will be queued and awaiting for 30 seconds won't be good enough to execute all the recovery tasks, but this functions will. try to kill them.

Reply:
Sure, the intention is that if we have a large number of tasks to execute, then it will be done in parallel and in batches (according to the available threads in the `pool`). However, the main thread would still wait for ALL recovery tasks to be done, and then shutdown the thread pool, because of L48:

`taskFutures.stream().map(CompletableFuture::join).collect(Collectors.toList());`

After L48, we can display the results, again because the large number of tasks is done, and then we can shutdown the thread pools safely. Let me know if I missed your points.

Reply:
I see.
I missed L48.
In that case, shall we call `shutdownNow` in function `shutdownAndAwaitTermination` instead of `shutdown`?
It is very common to use `shutdownNow` and `awaitTermination` together since `shutdown` only prevents the pool from accepting new tasks, and `shutdownNow` would try to stop the ongoing tasks.

Reply:
Sure, will change to `showdownNow`.